src/api/app/types.ts

```diff
@@ -101,11 +101,17 @@ export const SignalType = {
   System: "System",
 } as const;
 export type Signal = {
-  [SignalType.App]: [CellId, any];
+  [SignalType.App]: EncodedAppSignal;
   [SignalType.System]: unknown;
 };
+export type EncodedAppSignal = {
+  cell_id: CellId;
+  zome_name: string;
+  signal: Uint8Array;
+};
```

Comment:
What about introducing PermaLinks through to the Rust side code? Bumping into quite a few serialization issues ongoingly

Reply:
I'm not familiar with PermaLinks, what is that?

Reply:
oops, just meant like a github permalink... like this. 
<img width="586" alt="Screenshot 2023-01-18 at 9 08 02 PM" src="https://user-images.githubusercontent.com/1409121/213360140-25c69a9e-5584-4206-aa35-3df6074740aa.png">